pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -125,6 +125,15 @@ func ApplyTaskResults(spec *v1alpha1.TaskSpec) *v1alpha1.TaskSpec {
 	return ApplyReplacements(spec, stringReplacements, map[string][]string{})
 }
 
+// ApplyCredentialsPath applies a substitution of the key $(credentials.path) with the path that the creds-init
+// helper will write its credentials to.
+func ApplyCredentialsPath(spec *v1alpha1.TaskSpec, path string) *v1alpha1.TaskSpec {
```

Comment:
Is this only usable as internal ? (because I think it's gonna be invalid if using directly in a spec)

Reply:
Could you tell me more about it being invalid if used directly in spec?  It's intended for users to be able to work with as well.

Reply:
I think this might fail at validation saying `credential.path` is an invalid variable (not 100% sure but my feeling is that not having anything about it in the validation code means it will fail — maybe I am wrong though, a tests showing this would be nice :stuck_out_tongue_closed_eyes: )

Reply:
Aaaaah.  Lemme check.  Great catch.

Reply:
Initial manual test proved positive - I was able to use `$(credential.path)` in a TaskRun w/ embedded TaskSpec.  I didn't hit a validation error and the Task ran successfully.

I'm going to add a test for this right now to make sure it doesn't fail validation in future.

Reply:
Updated with tests for task and taskrun validations in both v1beta1 and v1alpha1 packages.